src/main/sensors/initialisation.c

```diff
@@ -734,7 +734,7 @@ static void detectMag(magSensor_e magHardwareToUse)
 
     if (magHardware == MAG_NONE && magHardwareToUse != MAG_DEFAULT && magHardwareToUse != MAG_NONE) {
         // Nothing was found and we have a forced sensor that isn't present.
-        magHardwareToUse = MAG_DEFAULT;
+        magHardwareToUse = MAG_NONE;
```

Comment:
This isn't really necessary, just replace both magHardwareToUse  assigment and "goto" statement with a "return".